### ISSUE TITLE
BSD ar doesn't support @linkerArgs.txt but Nim's compiler assumes it does

### ISSUE THREAD

**Issue by michaelsbradleyjr** [nim-lang/Nim#15589]
See: https://github.com/nim-lang/Nim/blob/51e3e0c7c43e82932d9bbae2e6ccc876d29b9fab/compiler/extccomp.nim#L918-L923

I have a GitHub Actions build that's failing on macOS but the build works correctly on my local macOS laptop. For the project I'm working on the `ar` command is 30010 chars in length locally but in GitHub Actions it's 33671, owing to differences in the absolute paths.

Per the compiler code linked above, when the length exceeds 32000, regardless of platform, Nim builds a `linkerArgs.txt` file and then invokes `ar` in this way:

```
ar @foo_linkerArgs.txt
```

However, BSD `ar`, which is what's available in macOS at `/usr/bin/ar` doesn't support that invocation, so it exits with error and that crashes the build.

**Comment by michaelsbradleyjr** [comment]
As a workaround `llvm-ar` can be used in place of `/usr/bin/ar` because `llvm-ar` supports the `@linkerArgs.txt` invocation.

On macOS in GitHub Actions I'm doing something like this:
```
# use `llvm-ar` instead of `ar` on macOS as a workaround for:
# https://github.com/nim-lang/Nim/issues/15589
if [[ ${{ matrix.platform.os }} = macos ]]; then
  mkdir -p "${HOME}/.local/bin"
  ln -f -s /usr/local/Cellar/llvm/*/bin/llvm-ar "${HOME}/.local/bin/ar"
  export PATH="${HOME}/.local/bin:${PATH}"
fi
```
And it works! 🎉

**Comment by Araq** [comment]
Is that even fixable on our side? Command line length limits are common.

**Comment by alaviss** [comment]
> Is that even fixable on our side? Command line length limits are common.

There's a stupid thing we can do... Just repeat `ar rs` or `ar qs` to keep appending files into the archive.

**Comment by michaelsbradleyjr** [comment]
Bitten again: https://github.com/status-im/nim-status/pull/87/checks?check_run_id=1656750955#step:9:6357

**Comment by michaelsbradleyjr** [comment]
Closed by #21381.
